client/layout/masterbar/logged-out.jsx

```diff
@@ -102,6 +102,15 @@ class MasterbarLoggedOut extends PureComponent {
 			return null;
 		}
 
+		/**
+		 * Hide signup from the screen when we have been sent to the login page from a redirect
+		 * by a service provider to authorize a Domain Connect template application.
+		 */
+		const redirectTo = get( currentQuery, 'redirect_to', '' );
+		if ( startsWith( redirectTo, '/domain-connect/authorize/' ) ) {
```

Comment:
Would probably be smart to extract to a helper.

Reply:
Good suggestion. Thanks!